src/server/pkg/storage/gc/client.go

```diff
@@ -24,10 +33,12 @@ var (
 //  * Source - the source of the reference, this may be a temporary id, chunk id, or a
 //    semantic name.
 //  * Chunk - the target chunk being referenced.
+//  * ExpiresAt - the time after which a temporary chunk will be destroyed
 type Reference struct {
-	Sourcetype string
+	Sourcetype SourceType
 	Source     string
 	Chunk      string
+	ExpiresAt  *time.Time
```

Comment:
Why is this a pointer?

Reply:
It looks like gorm is operating on a pointer in some queries and a struct in others.

Reply:
In our model it's optional. When interacting with the database we use `sql.NullTime`